support-data-rest/src/test/java/uk/ac/ebi/uniprot/crossref/controller/CrossRefGetIdControllerIT.java

```diff
@@ -0,0 +1,179 @@
+                    .resultMatcher(jsonPath("$.db_url", is("http://tigrfams.jcvi.org/cgi-bin/HmmReportPage.cgi?acc=%s")))
+                    .resultMatcher(jsonPath("$.pub_med_id", is("17151080")))
+                    .resultMatcher(jsonPath("$.accession", is(ACCESSION)))
+                    .resultMatcher(jsonPath("$.abbrev", is("TIGRFAMs")))
+                    .resultMatcher(jsonPath("$.reviewed_protein_count", is(10)))
+                    .resultMatcher(jsonPath("$.category", is("Family and domain databases")))
+                    .resultMatcher(jsonPath("$.doi_id", is("10.1093/nar/gkl1043")))
+                    .build();
+        }
+
+        @Override
+        public GetIdParameter invalidIdParameter() {
+            return GetIdParameter.builder().id("INVALID")
+                    .resultMatcher(jsonPath("$.url", not(isEmptyOrNullString())))
+                    .resultMatcher(jsonPath("$.messages.*", contains("The cross ref id value should be in the form of DB-XXXX")))
+                    .build();
+        }
+
+        @Override
+        public GetIdParameter nonExistentIdParameter() {
+            return GetIdParameter.builder().id("DB-0000")
+                    .resultMatcher(jsonPath("$.url", not(isEmptyOrNullString())))
+                    .resultMatcher(jsonPath("$.messages.*", contains("Resource not found")))
+                    .build();
+        }
+
+        @Override
+        public GetIdParameter withFilterFieldsParameter() {
+            return GetIdParameter.builder().id(ACCESSION).fields("accession,category,unreviewed_protein_count")
+                    .resultMatcher(jsonPath("$.accession", is(ACCESSION)))
+                    .resultMatcher(jsonPath("$.category", is("Family and domain databases")))
+                    .resultMatcher(jsonPath("$.unreviewed_protein_count", is(5)))
+                    .resultMatcher(jsonPath("$.name").doesNotExist())
+                    .resultMatcher(jsonPath("$.reviewed_protein_count").doesNotExist())
+                    .build();
+        }
+
+        @Override
+        public GetIdParameter withInvalidFilterParameter() {
+            return GetIdParameter.builder().id(ACCESSION).fields("invalid")
+                    .resultMatcher(jsonPath("$.url", not(isEmptyOrNullString())))
+                    .resultMatcher(jsonPath("$.messages.*", contains("Invalid fields parameter value 'invalid'")))
+                    .build();
+        }
+    }
+
+    static class CrossRefGetIdContentTypeParamResolver extends AbstractGetIdContentTypeParamResolver {
+
+        @Override
+        public GetIdContentTypeParam idSuccessContentTypesParam() {
+            return GetIdContentTypeParam.builder()
+                    .id(ACCESSION)
+                    .contentTypeParam(ContentTypeParam.builder()
+                            .contentType(MediaType.APPLICATION_JSON)
+                            .resultMatcher(jsonPath("$.link_type", is("Explicit")))
```

Comment:
Would it be possible for these names to be camel case (same for the ones below)?

Reply:
Will fix as a separate issue fir disease and cross ref responses.